drake/core/Function.h

```diff
@@ -104,46 +106,54 @@ namespace Drake {
     }
   };
 
-  class Function {
-  public:
-    Function() : relation(InputOutputRelation::Form::ARBITRARY) {};
-    Function(InputOutputRelation::Form f) : relation(f) {};
-    virtual ~Function() {};
+  template <typename ScalarType> using VecIn = Eigen::Ref< Eigen::Matrix<ScalarType, Eigen::Dynamic, 1> const >;
+  template <typename ScalarType> using VecOut = Eigen::Matrix<ScalarType, Eigen::Dynamic, 1>;
```

Comment:
can we reuse the typedefs in core/Vector.h instead?  or add to them if the Ref is missing?